pyproj/crs.py

```diff
@@ -249,59 +283,100 @@ def from_epsg(cls, code):
         Parameters
         ----------
         code : int or str
-            An EPSG code. Strings will be converted to integers.
+            An EPSG code.
+
+        Returns
+        -------
+        CRS
+        """
+        return cls(_from_epsg(code))
+
+    @classmethod
+    def from_proj4(cls, in_proj_string):
+        """Make a CRS from a PROJ string
 
-        Notes
-        -----
-        The input code is not validated against an EPSG database.
+        Parameters
+        ----------
+        in_proj_string : str
+            A PROJ string.
 
         Returns
         -------
         CRS
         """
-        return cls.from_authority("epsg", code)
+        if is_wkt(in_proj_string) or "=" not in in_proj_string:
+            raise CRSError("Invalid PROJ string: {}".format(in_proj_string))
+        return cls(_from_string(in_proj_string))
 
     @classmethod
-    def from_string(cls, proj_string):
-        """Make a CRS from an EPSG, PROJ, or WKT string
+    def from_wkt(cls, in_wkt_string):
+        """Make a CRS from a WKT string
 
         Parameters
         ----------
-        proj_string : str
+        in_wkt_string : str
+            A WKT string.
+
+        Returns
+        -------
+        CRS
+        """
+        if not is_wkt(in_wkt_string):
+            raise CRSError("Invalid WKT string: {}".format(in_wkt_string))
+        return cls(_from_string(in_wkt_string))
+
+    @classmethod
+    def from_string(cls, in_crs_string):
+        """Make a CRS from:
+
+        Initialize a CRS class instance with:
+         - PROJ string 
+         - JSON string with PROJ parameters
+         - CRS WKT string
+         - An authority string [i.e. 'epsg:4326']
+
+        Parameters
+        ----------
+        in_crs_string : str
             An EPSG, PROJ, or WKT string.
 
         Returns
         -------
         CRS
         """
-        if not proj_string:
-            raise CRSError("CRS is empty or invalid: {!r}".format(proj_string))
+        return cls(_from_string(in_crs_string))
 
-        elif proj_string.strip().upper().startswith("EPSG:"):
-            auth, val = proj_string.strip().split(":")
-            if not val:
-                raise CRSError("Invalid CRS: {!r}".format(proj_string))
-            return cls.from_epsg(val)
+    def to_string(self):
+        """Convert the CRS to a string.
```

Comment:
My first reaction was: what's the purpose of this? But actually, this could be quite nice. It kind of gives  the "best possible string representation": if possible a "EPSG:.." string, otherwise the string it is based on.

Reply:
I had a similar thought process :-)